src/libraries/System.IO/tests/Stream/Stream.CopyToSpanTests.cs

```diff
@@ -0,0 +1,528 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class StreamCopyToSpanTests
+    {
+        [Fact]
+        public void CopyTo_InvalidArgsThrows()
+        {
+            using Stream s = new MemoryStream();
+
+            AssertExtensions.Throws<ArgumentNullException>("callback", () => s.CopyTo(null, null, 0));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("bufferSize", () => s.CopyTo((_, __) => { }, null, 0));
+
+            AssertExtensions.Throws<ArgumentNullException>("callback", () => s.CopyToAsync(null, null, 0, default));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("bufferSize", () => s.CopyToAsync((_, __, ___) => default, null, 0, default));
+        }
+
+        [Fact]
+        public void CopyToAsync_PrecanceledToken_Cancels()
+        {
+            using var src = new MemoryStream();
+            Assert.Equal(TaskStatus.Canceled, src.CopyToAsync((_, __, ___) => default, null, 4096, new CancellationToken(true)).Status);
+        }
+
+        [Theory]
+        [MemberData(nameof(CopyTo_TestData))]
+        public async Task CopyToAsync_CancellationToken_Propagated(MemoryStream input)
+        {
+            using var src = input;
+            src.WriteByte(0);
+            src.Position = 0;
+
+            var cancellationToken = new CancellationToken();
```

Comment:
`CancellationToken` is a struct.  This is the same as `CancellationToken cancellationToken = default;`, which is probably not what you want.  Instead, you probably want:
```suggestion
            CancellationToken cancellationToken = new CancellationTokenSource().Token;
```